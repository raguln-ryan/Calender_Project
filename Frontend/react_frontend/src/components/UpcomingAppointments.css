.upcoming-appointments ul {
  list-style: none;
  padding: 0;
}

.appointment-card {
  margin: 10px 0;
  padding: 12px;
  border-radius: 10px;
  color: #fff; /* white text on colored background */
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.15);
}

.appointment-info {
  margin-bottom: 10px;
}

.type-badge {
  display: inline-block;
  padding: 2px 8px;
  margin: 5px 0;
  border-radius: 6px;
  background: rgba(255, 255, 255, 0.25);
  font-size: 12px;
  font-weight: bold;
}
.actions.below {
  display: flex;
  justify-content: center; /* Centers horizontally */
  gap: 10px; /* Adds space between buttons */
  margin-top: 10px; /* Optional, to add spacing */
}
.edit-btn,
.delete-btn {
  border: none;
  padding: 6px 12px;
  border-radius: 6px;
  cursor: pointer;
  font-weight: bold;
}

.edit-btn {
  background-color: white;
  color: black;
}

.delete-btn {
  background-color: white;
  color: black;
}
.upcoming-appointments {
  max-width: 100%;
  margin: 0 auto;
  padding: 12px;
  display: flex;
  flex-direction: column;
  height: 100%;
}

/* ðŸ”¹ Sticky search bar at the top */
.search-container {
  position: sticky;
  top: 0;
  background: #fff;
  z-index: 10;
  padding: 6px 0;
}

.search-input {
  width: 100%;
  padding: 8px;
  font-size: 1rem;
  border: 1px solid #ccc;
  border-radius: 6px;
}

/* ðŸ”¹ Scrollable appointment list */
.appointments-list {
  flex: 1;
  max-height: 400px; /* adjust as needed */
  overflow-y: auto;
  margin-top: 10px;
  padding-right: 6px;
}

/* Appointment card styles */
.appointment-card {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 12px;
  padding: 12px;
  border-radius: 6px;
  color: #fff;
  font-size: 0.95rem;
}

.appointment-info {
  flex: 1;
}

.actions.below {
  display: flex;
  flex-direction: column;
  gap: 6px;
}

/* ðŸ”¹ Mobile responsiveness */
@media (max-width: 768px) {
  .appointment-card {
    flex-direction: column;
    align-items: flex-start;
  }

  .actions.below {
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
    margin-top: 8px;
  }
}