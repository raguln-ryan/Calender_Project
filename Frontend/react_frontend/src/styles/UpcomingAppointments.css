/* ==============================
   Upcoming Appointments Sidebar
   ============================== */

/* Sidebar */
.upcoming-appointments {
  position: fixed;
  top: 0;
  left: -100%;
  width: 100%; /* Mobile default */
  max-width: 400px; /* Max width for iPad/large devices */
  height: 100%;
  background: #fff;
  color: #000;
  box-shadow: 2px 0 6px rgba(0, 0, 0, 0.2);
  overflow: hidden; /* Prevent page scroll when sidebar open */
  transition: left 0.3s ease-in-out;
  z-index: 2000;
  padding: 1rem;
  box-sizing: border-box;
}

.upcoming-appointments.open {
  left: 0;
}

/* Overlay */
.overlay {
  display: none;
}

.overlay.show {
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.4);
  z-index: 1900;
}

/* Hamburger Button */
.hamburger-btn {
  display: none;
  position: fixed;
  top: 15px;
  left: 15px;
  z-index: 2100;
  font-size: 24px;
  background: #3f51b5;
  color: #fff;
  border: none;
  border-radius: 4px;
  padding: 8px 12px;
  cursor: pointer;
}

/* Appointment List */
.upcoming-appointments ul {
  list-style: none;
  padding: 0;
  margin: 0;
}

.appointment-card {
  margin: 12px 0;
  padding: 14px;
  border-radius: 10px;
  background: #3f51b5;
  color: #fff;
  display: flex;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
}

.appointment-info {
  margin-bottom: 10px;
  word-wrap: break-word;
}

.type-badge {
  display: inline-block;
  padding: 2px 8px;
  margin: 5px 0;
  border-radius: 6px;
  background: rgba(255, 255, 255, 0.25);
  font-size: 12px;
  font-weight: bold;
}

.actions.below {
  display: flex;
  gap: 10px;
  justify-content: center;
  flex-wrap: wrap;
  margin-top: 10px;
}

.edit-btn,
.delete-btn {
  border: none;
  padding: 6px 12px;
  border-radius: 6px;
  cursor: pointer;
  font-weight: bold;
  background: #fff;
  color: #000;
  min-width: 70px;
  text-align: center;
}

/* Search + Filters */
.search-container {
  position: sticky;
  top: 0;
  background: #fff;
  z-index: 10;
  padding: 8px 0;
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.search-input,
.date-picker-input input {
  width: 100%;
  padding: 10px 15px;
  border-radius: 50px;
  border: 1px solid #ccc;
  font-size: 14px;
  outline: none;
  background: #fff;
  color: #000;
}

.date-picker-input {
  position: relative;
  display: flex;
  align-items: center;
}

.clear-date-btn {
  position: absolute;
  right: 12px;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  border: none;
  background: #000;
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 14px;
  cursor: pointer;
}

.clear-date-btn:hover {
  background: #333;
}

/* Scrollable appointment list inside sidebar */
.appointments-scroll {
  /* Use dynamic viewport height minus top padding & search height */
  max-height: calc(100vh - 140px);
  overflow-y: auto;
  scroll-behavior: smooth;
}

.appointments-scroll::-webkit-scrollbar {
  width: 6px;
}

.appointments-scroll::-webkit-scrollbar-thumb {
  background-color: rgba(0,0,0,0.3);
  border-radius: 3px;
}

/* ==============================
   Responsive for Tablets & Mobiles
   ============================== */
@media (max-width: 767px) {
  /* Mobile */
  .hamburger-btn {
    display: block;
  }

  .upcoming-appointments {
    width: 100%;
  }

  .section-title {
    display: none;
  }

  .appointment-card {
    flex-direction: column;
    align-items: flex-start;
  }

  .actions.below {
    flex-direction: column;
    width: 100%;
  }
}

/* iPad Portrait */
@media (min-width: 768px) and (max-width: 1024px) {
  .hamburger-btn {
    display: block;
    top: 20px;
    left: 20px;
  }

  .upcoming-appointments {
    width: 80%;
  }
}

/* iPad Landscape */
@media (min-width: 1025px) and (max-width: 1280px) {
  .hamburger-btn {
    display: block;
    top: 20px;
    left: 20px;
  }

  .upcoming-appointments {
    width: 65%;
  }
}

/* Desktop: always show sidebar */
@media (min-width: 1281px) {
  .upcoming-appointments {
    position: relative;
    left: 0;
    width: 350px;
    height: auto;
    box-shadow: none;
  }

  .hamburger-btn {
    display: none;
  }

  .section-title {
    display: block;
  }
}

/* Dark Mode */
.dark-mode .upcoming-appointments {
  background: #121212;
  color: #fff;
}

.dark-mode .search-container {
  background: #121212;
}

.dark-mode .search-input,
.dark-mode .date-picker-input input {
  background: #1e1e1e;
  color: #fff;
  border: 1px solid #555;
}

.dark-mode .clear-date-btn {
  background: #fff;
  color: #000;
}

.dark-mode .clear-date-btn:hover {
  background: #ddd;
}

.dark-mode .overlay.show {
  background: rgba(255, 255, 255, 0.15);
}
