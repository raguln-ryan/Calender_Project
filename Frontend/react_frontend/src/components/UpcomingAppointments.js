import React from 'react';
import './UpcomingAppointments.css';

const UpcomingAppointments = ({ appointments }) => {
  // Filter and sort appointments to get upcoming ones
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  
  const upcomingAppointments = appointments
    .filter(appointment => {
      const appointmentDate = new Date(appointment.date);
      return appointmentDate >= today;
    })
    .sort((a, b) => new Date(a.date) - new Date(b.date))
    .slice(0, 5); // Show only next 5 appointments
  
  return (
    <div className="upcoming-appointments">
      <h3>Upcoming Appointments</h3>
      {upcomingAppointments.length > 0 ? (
        <ul className="appointment-list">
          {upcomingAppointments.map((appointment) => {
            const appointmentDate = new Date(appointment.date);
            return (
              <li 
                key={appointment.id} 
                className={`appointment-item appointment-${appointment.type || 'other'}`}
              >
                <div className="appointment-date">
                  {appointmentDate.toLocaleDateString()}
                </div>
                <div className="appointment-time">
                  {appointment.time}
                </div>
                <div className="appointment-title">
                  {appointment.title}
                </div>
              </li>
            );
          })}
        </ul>
      ) : (
        <p className="no-appointments">No upcoming appointments</p>
      )}
    </div>
  );
};

export default UpcomingAppointments;
