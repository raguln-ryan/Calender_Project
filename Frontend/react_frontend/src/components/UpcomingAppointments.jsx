import React, { useState, useEffect } from "react";
import { getUpcomingAppointments, deleteAppointment } from "../services/api";
import "../styles/UpcomingAppointments.css";

const UpcomingAppointments = ({ refreshTrigger, onEdit, onDelete }) => {
  const [appointments, setAppointments] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState("");
  const [searchQuery, setSearchQuery] = useState("");
  const [selectedDate, setSelectedDate] = useState("");
  const [isOpen, setIsOpen] = useState(false);
  const [windowWidth, setWindowWidth] = useState(window.innerWidth);

  const fetchAppointments = async () => {
    try {
      setLoading(true);
      const data = await getUpcomingAppointments(30);
      const upcoming = data?.filter((a) => !a.completed) || [];
      setAppointments(upcoming);
      setLoading(false);
    } catch (err) {
      console.error(err);
      setError("Failed to fetch upcoming appointments.");
      setLoading(false);
    }
  };

  useEffect(() => {
    fetchAppointments();
  }, [refreshTrigger]);

  // Handle window resize to detect iPad/mobile
  useEffect(() => {
    const handleResize = () => setWindowWidth(window.innerWidth);
    window.addEventListener("resize", handleResize);
    return () => window.removeEventListener("resize", handleResize);
  }, []);

  const handleDelete = async (id) => {
    const confirmDelete = window.confirm(
      "Are you sure you want to delete this appointment?"
    );
    if (!confirmDelete) return;

    try {
      await deleteAppointment(id);
      fetchAppointments();
      if (onDelete) onDelete();
    } catch (err) {
      console.error("Error deleting appointment:", err);
      alert("Failed to delete appointment. Please try again.");
    }
  };

  // Filter appointments
  const filteredAppointments = appointments.filter((a) => {
    const title = a?.title?.toLowerCase() || "";
    const type = a?.type?.toLowerCase() || "";
    const matchesSearch =
      title.includes(searchQuery.toLowerCase()) ||
      type.includes(searchQuery.toLowerCase());

    if (!selectedDate) return matchesSearch;

    const appointmentDate = new Date(a.startTime);
    const selected = new Date(selectedDate);

    return (
      matchesSearch &&
      appointmentDate.getFullYear() === selected.getFullYear() &&
      appointmentDate.getMonth() === selected.getMonth() &&
      appointmentDate.getDate() === selected.getDate()
    );
  });

  // Detect if device is tablet
  const isTablet = windowWidth >= 768 && windowWidth <= 1280;

  return (
    <>
      {/* Hamburger button: show only when sidebar is closed */}
      {!isOpen && (
        <button
          className="hamburger-btn"
          onClick={() => setIsOpen(true)}
        >
          ‚ò∞
        </button>
      )}

      {/* Show Upcoming Appointments Button (iPad only) */}
      {isTablet && !isOpen && (
        <button
          className="show-appointments-btn"
          onClick={() => setIsOpen(true)}
        >
          Show Upcoming Appointments
        </button>
      )}

      {/* Overlay */}
      {isOpen && <div className="overlay show" onClick={() => setIsOpen(false)} />}

      {/* Sidebar Drawer */}
      <div className={`upcoming-appointments ${isOpen ? "open" : ""}`}>
        <h3 className="section-title">üìÖ Upcoming Appointments</h3>

        {/* Search + Date Picker */}
        <div className="search-container">
          <div className="date-picker-input">
            <input
              type="date"
              value={selectedDate}
              onChange={(e) => setSelectedDate(e.target.value)}
            />
            {selectedDate && (
              <button
                className="clear-date-btn"
                onClick={() => setSelectedDate("")}
              >
                ‚úñ
              </button>
            )}
          </div>

          <input
            type="text"
            placeholder="Search appointments..."
            value={searchQuery}
            onChange={(e) => setSearchQuery(e.target.value)}
            className="search-input"
          />
        </div>

        {/* Appointment List with scroll */}
        <div className="appointments-scroll">
          {loading ? (
            <p>Loading...</p>
          ) : error ? (
            <p className="error-text">{error}</p>
          ) : filteredAppointments.length === 0 ? (
            <p>No appointments found.</p>
          ) : (
            <ul className="appointments-list">
              {filteredAppointments.map((a) => (
                <li
                  key={a.id}
                  className="appointment-card"
                  style={{
                    backgroundColor: a.color || "#3f51b5",
                    borderLeft: `8px solid ${a.color || "#3f51b5"}`,
                  }}
                >
                  <div className="appointment-info">
                    <strong className="appointment-title">{a.title}</strong>
                    <br />
                    <span className="type-badge">{a.type || "General"}</span>
                    <br />
                    <span className="appointment-time">
                      {new Date(a.startTime).toLocaleString()} ‚Äì{" "}
                      {new Date(a.endTime).toLocaleTimeString()}
                    </span>
                  </div>

                  <div className="actions below">
                    <button
                      className="edit-btn"
                      onClick={() => onEdit && onEdit(a)}
                    >
                      ‚úèÔ∏è Edits
                    </button>
                    <button
                      className="delete-btn"
                      onClick={() => handleDelete(a.id)}
                    >
                      üóëÔ∏è Delete
                    </button>
                  </div>
                </li>
              ))}
            </ul>
          )}
        </div>
      </div>
    </>
  );
};

export default UpcomingAppointments;
