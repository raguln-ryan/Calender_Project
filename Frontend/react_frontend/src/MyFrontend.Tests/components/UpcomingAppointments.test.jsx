// src/MyFrontend.Tests/components/UpcomingAppointments.full.test.jsx
import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import UpcomingAppointments from "../../components/UpcomingAppointments";

const mockAppointments = [
  { 
    id: 1, 
    title: "Meeting", 
    type: "Call", 
    startTime: "2025-09-25T10:00:00.000Z", 
    endTime: "2025-09-25T11:00:00.000Z", 
    color: "#2196F3", 
    completed: false,
    description: "Team sync"
  },
  { 
    id: 2, 
    title: "Project", 
    type: "Meeting", 
    startTime: "2025-09-26T09:00:00.000Z", 
    endTime: "2025-09-26T10:00:00.000Z", 
    color: "#4CAF50", 
    completed: true,
    description: "Sprint planning"
  },
  { 
    id: 3, 
    title: "NoType", 
    startTime: "2025-09-27T09:00:00.000Z", 
    endTime: "2025-09-27T10:00:00.000Z", 
    color: "", 
    completed: false 
  },
  {
    id: 4,
    title: "Past Event",
    type: "Event",
    startTime: "2023-01-01T09:00:00.000Z",
    endTime: "2023-01-01T10:00:00.000Z",
    color: "#FF5733",
    completed: true
  }
];

const defaultProps = {
  refreshTrigger: 0,
  onEdit: jest.fn(),
  onDelete: jest.fn()
};

describe("UpcomingAppointments Complete Coverage", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    window.confirm = jest.fn().mockReturnValue(true);
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  // Basic rendering tests
  test("renders header and search input", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    expect(screen.getByText(/Upcoming Appointments/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Search appointments/i)).toBeInTheDocument();
  });

  test("renders appointments grouped by date", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    expect(screen.getByText("9/25/2025")).toBeInTheDocument();
    expect(screen.getByText("9/26/2025")).toBeInTheDocument();
    expect(screen.getByText("9/27/2025")).toBeInTheDocument();
  });

  test("renders appointment badges correctly", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    expect(screen.getByText("Call")).toBeInTheDocument();
    expect(screen.getByText("Meeting")).toHaveClass("badge");
  });

  // Search functionality tests
  test("filters appointments by title search", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    fireEvent.change(screen.getByPlaceholderText(/Search appointments/i), { target: { value: "Project" } });
    expect(screen.queryByText("Meeting")).not.toBeInTheDocument();
    expect(screen.getByText("Project")).toBeInTheDocument();
  });

  test("filters appointments by type search", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    fireEvent.change(screen.getByPlaceholderText(/Search appointments/i), { target: { value: "Call" } });
    expect(screen.getByText("Meeting")).toBeInTheDocument();
    expect(screen.queryByText("Project")).not.toBeInTheDocument();
  });

  test("case-insensitive search", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    fireEvent.change(screen.getByPlaceholderText(/Search appointments/i), { target: { value: "PROJECT" } });
    expect(screen.getByText("Project")).toBeInTheDocument();
  });

  test("search with special characters", () => {
    const specialAppt = [{ ...mockAppointments[0], title: "Meeting @ 10:00" }];
    render(<UpcomingAppointments {...defaultProps} appointments={specialAppt} />);
    fireEvent.change(screen.getByPlaceholderText(/Search appointments/i), { target: { value: "@" } });
    expect(screen.getByText("Meeting @ 10:00")).toBeInTheDocument();
  });

  test("clears search filter", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    const searchInput = screen.getByPlaceholderText(/Search appointments/i);
    fireEvent.change(searchInput, { target: { value: "Project" } });
    expect(screen.queryByText("Meeting")).not.toBeInTheDocument();
    
    fireEvent.change(searchInput, { target: { value: "" } });
    expect(screen.getByText("Meeting")).toBeInTheDocument();
  });

  // Date filter tests
  test("filters appointments by selected date", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    const dateInput = screen.getByLabelText(/Filter by date/i);
    fireEvent.change(dateInput, { target: { value: "2025-09-25" } });
    expect(screen.getByText("Meeting")).toBeInTheDocument();
    expect(screen.queryByText("Project")).not.toBeInTheDocument();
  });

  test("clears date filter with clear button", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    const dateInput = screen.getByLabelText(/Filter by date/i);
    fireEvent.change(dateInput, { target: { value: "2025-09-25" } });
    
    const clearBtn = screen.getByRole("button", { name: /Clear date/i });
    fireEvent.click(clearBtn);
    
    expect(screen.getByText("Meeting")).toBeInTheDocument();
    expect(screen.getByText("Project")).toBeInTheDocument();
  });

  test("combines search and date filters", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    fireEvent.change(screen.getByPlaceholderText(/Search appointments/i), { target: { value: "Meeting" } });
    fireEvent.change(screen.getByLabelText(/Filter by date/i), { target: { value: "2025-09-25" } });
    
    expect(screen.getByText("Meeting")).toBeInTheDocument();
    expect(screen.queryByText("Project")).not.toBeInTheDocument();
    expect(screen.queryByText("NoType")).not.toBeInTheDocument();
  });

  // Sidebar toggle tests
  test("toggles sidebar with hamburger menu", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    const hamburgerBtn = screen.getByRole("button", { name: /Toggle menu/i });
    
    // Open sidebar
    fireEvent.click(hamburgerBtn);
    expect(screen.getByTestId("sidebar")).toHaveClass("open");
    
    // Close sidebar
    fireEvent.click(hamburgerBtn);
    expect(screen.getByTestId("sidebar")).not.toHaveClass("open");
  });

  test("closes sidebar with overlay click", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    fireEvent.click(screen.getByRole("button", { name: /Toggle menu/i }));
    
    const overlay = screen.getByTestId("overlay");
    fireEvent.click(overlay);
    
    expect(screen.getByTestId("sidebar")).not.toHaveClass("open");
  });

  test("show appointments button opens sidebar", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    const showBtn = screen.getByRole("button", { name: /Show Appointments/i });
    fireEvent.click(showBtn);
    expect(screen.getByTestId("sidebar")).toHaveClass("open");
  });

  // Action button tests
  test("edit button triggers onEdit with correct appointment", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    const editBtns = screen.getAllByRole("button", { name: /Edit/i });
    fireEvent.click(editBtns[0]);
    expect(defaultProps.onEdit).toHaveBeenCalledWith(mockAppointments[0]);
  });

  test("delete button shows confirmation and triggers onDelete", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    const deleteBtns = screen.getAllByRole("button", { name: /Delete/i });
    fireEvent.click(deleteBtns[0]);
    
    expect(window.confirm).toHaveBeenCalledWith("Are you sure you want to delete this appointment?");
    expect(defaultProps.onDelete).toHaveBeenCalledWith(mockAppointments[0].id);
  });

  test("delete cancellation doesn't trigger onDelete", () => {
    window.confirm = jest.fn().mockReturnValue(false);
    render(<UpcomingAppointments {...defaultProps} appointments={mockAppointments} />);
    const deleteBtns = screen.getAllByRole("button", { name: /Delete/i });
    fireEvent.click(deleteBtns[0]);
    
    expect(defaultProps.onDelete).not.toHaveBeenCalled();
  });

  // Edge cases
  test("handles empty appointments array", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={[]} />);
    expect(screen.getByText(/No upcoming appointments/i)).toBeInTheDocument();
  });

  test("handles null appointments prop", () => {
    render(<UpcomingAppointments {...defaultProps} appointments={null} />);
    expect(screen.getByText(/No upcoming appointments/i)).toBeInTheDocument();
  });

  test("handles undefined appointments prop", () => {
    render(<UpcomingAppointments {...defaultProps} appointments
