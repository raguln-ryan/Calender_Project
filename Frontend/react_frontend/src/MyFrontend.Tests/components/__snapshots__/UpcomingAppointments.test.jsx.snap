// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpcomingAppointments Component snapshot matches 1`] = `
<div>
  <button
    class="hamburger-btn"
  >
    â˜°
  </button>
  <button
    class="show-appointments-btn"
  >
    Show Upcoming Appointments
  </button>
  <div
    class="upcoming-appointments "
  >
    <h3
      class="section-title"
    >
      ðŸ“… Upcoming Appointments
    </h3>
    <div
      class="search-container sticky-header"
    >
      <div
        class="date-picker-input"
      >
        <input
          min="2025-09-25"
          type="date"
          value=""
        />
      </div>
      <input
        class="search-input"
        placeholder="Search appointments..."
        type="text"
        value=""
      />
    </div>
    <div
      class="appointments-scroll with-header"
    >
      <p>
        Loading...
      </p>
    </div>
  </div>
</div>
`;

exports[`UpcomingAppointments Full Coverage snapshot test 1`] = `
<div>
  <button
    class="hamburger-btn"
  >
    â˜°
  </button>
  <button
    class="show-appointments-btn"
  >
    Show Upcoming Appointments
  </button>
  <div
    class="upcoming-appointments "
  >
    <h3
      class="section-title"
    >
      ðŸ“… Upcoming Appointments
    </h3>
    <div
      class="search-container sticky-header"
    >
      <div
        class="date-picker-input"
      >
        <input
          min="2025-09-25"
          type="date"
          value=""
        />
      </div>
      <input
        class="search-input"
        placeholder="Search appointments..."
        type="text"
        value=""
      />
    </div>
    <div
      class="appointments-scroll with-header"
    >
      <p>
        Loading...
      </p>
    </div>
  </div>
</div>
`;
